Storage/FirebaseStorageUI/FUIStorageImageLoader.m

```diff
@@ -85,27 +86,29 @@ - (BOOL)canRequestImageForURL:(NSURL *)url {
     size = self.defaultMaxImageSize;
   }
   // Download the image from Firebase Storage
-  
-  FIRStorageDownloadTask * download = [storageRef dataWithMaxSize:size
-                                                       completion:^(NSData * _Nullable data, NSError * _Nullable error) {
-                                                         if (error) {
-                                                           dispatch_main_async_safe(^{
-                                                             if (completedBlock) {
-                                                               completedBlock(nil, nil, error, YES);
-                                                             }
-                                                           });
-                                                           return;
-                                                         }
-                                                         // Decode the image with data
-                                                         dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
-                                                           UIImage *image = SDImageLoaderDecodeImageData(data, url, options, context);
-                                                           dispatch_main_async_safe(^{
-                                                             if (completedBlock) {
-                                                               completedBlock(image, data, nil, YES);
-                                                             }
-                                                           });
-                                                         });
-                                                       }];
+  // Each download task use independent serial coder queue, to ensure callback in order during prorgessive decoding
+  NSOperationQueue *coderQueue = [NSOperationQueue new];
```

Comment:
It looks like a new `coderQueue` is allocated on each invocation of this method, but never retained anywhere. Is this intentional?

Reply:
Bug seems I reset the wrong code. The dispatch_async should be replaced with coderQueue addOperationWithBlock:...Wait me fix this

Reply:
The lifetime is associated to the completion block, and the completion block lifetime is associated to `FIRStorageDownloadTask`. This implements the design I want : Each download task use independent serial coder queue

Reply:
Sounds good.

Reply:
Or, if you think it's hard to find, we can change this into an associated object of `FIRStorageDownloadTask`, like

```
@interface FIRStorageDownloadTask (StorageUI)
@property (nonatomic) NSOperationQueue *coderQueue;
@end
@implementation FIRStorageDownloadTask (StorageUI)
- (NSOperationQueue *)coderQueue {
    NSOperationQueue *coderQueue = objc_getAsscoaitedObject(self, @selector(coderQueue));
    if (!coderQueue) {
        coderQueue = [NSOperationQueue new];
        coderQueue.maxConcurrentOperationCount = 1;
        objc_setAssociatedObject(self, @selector(coderQueue), coderQueue, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
    } 
    return coderQueue;
}
@end
```

Reply:
I'm ok with the way the code is, just wanted to clarify 👍